docs/src/components/main-nav.tsx

```diff
@@ -30,9 +30,9 @@ export function MainNav(props: { items: NavItem[] }) {
                   key={index}
                   href={item.href}
                   className={cn(
-                    "text-muted-foreground flex items-center justify-center rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:opacity-50 disabled:pointer-events-none ring-offset-background hover:bg-accent hover:text-accent-foreground h-9 px-3",
+                    "text-foreground/60 text-sm font-medium transition-colors hover:text-foreground/80",
                     item.disabled && "cursor-not-allowed opacity-80",
-                    item.href === pathname && "text-foreground"
+                    item.href.startsWith(`/${segment}`) && "text-foreground"
```

Comment:
if we reached the fallback state of `/` here, we'd get `item.startsWith("//")` which won't match anything

Reply:
i get what you are saying but, if the pathname is "//" it will be redirected to "/". so I think this won't cause any problems. i am not 100% sure. so you call it

Reply:
ya. i think we should not have any fallback. i checked the nextjs docs and they recommend using it this way `const segment = useSelectedLayoutSegment();` This will be correct, just the strict typescript checking will throw an error, and we won't have any issues on the site.

Reply:
the hook returns `null` if there isn't a segment so it's not the strict typescript checking that causes the type to be null in some cases.

Reply:
okay, what do you think about this ```item.href.startsWith(`/${segment}` ?? "")``` the strict typescript checking will still throw an error. but I think we can ignore it this time, what do you think?

![image](https://github.com/t3-oss/t3-env/assets/109742866/c4d52b35-9342-4c76-99d1-46a842eb5f08)

Reply:
i agree that this task is a skill issue for me.